src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,535 @@
+    slug,
+    description,
+    date: new Date(date).toLocaleDateString()
+  };
+});
+---
+<html lang="en">
+  <head>
+    <title>My Blog</title>
+  </head>
+  <body>
+    <h1>My Blog</h1>
+    <ul>
+      {posts.map((post) => (
+        <li>
+          <a href={`/posts/${slug}/`}>
+            <h2>{title}</h2>
+          </a>
+          <time>{date}</time>
+          <p>{description}</p>
+        </li>
+      ))}
+    </ul>
+  </body>
+</html>
+```
+
+### Dynamic routes
+
+#### Static site generation
+
+To generate our blog posts pages statically, we will use [dynamic routes](/en/core-concepts/routing/#dynamic-routes) and the `getStaticPaths()` function. This function will be called at build time to generate the list of paths that will be rendered to HTML.
+
+Create a new file named `[slug].astro` in `src/pages/posts/` and import the `blogPost` interface and `contentfulClient` from `src/lib/contentful.ts`. 
+
+Inside the `getStaticPaths()` function, fetch all the entries with a content type of `blogPost` using the `getEntries` method from `contentfulClient`. You can pass the `blogPost` interface to the `getEntries` method to type your response. 
+
+```astro title="src/pages/posts/[slug].astro"
+---
+import { contentfulClient } from "../../lib/contentful";
+import type { blogPostFields } from "../../lib/contentful";
+
+export async function getStaticPaths() {
+  const entries = await contentfulClient.getEntries<blogPost>({
+    content_type: "blogPost",
+  });
+}
+---
+```
+
+The items property of `entries` contains a list of blog posts. We will use the `map` method to iterate over the list and return a list of pages. Each page will be an object with a `params` and `props` property. The `params` property will be used to generate the URL of the page and the `props` property will be passed to the page component as props.
+
+```astro title="src/pages/posts/[slug].astro" ins={3,11-19}
+---
+import { contentfulClient } from "../../lib/contentful";
+import { documentToHtmlString } from "@contentful/rich-text-html-renderer";
+import type { blogPostFields } from "../../lib/contentful";
+
+export async function getStaticPaths() {
+  const entries = await contentfulClient.getEntries<blogPostFields>({
+    content_type: "blogPost",
+  });
+
+  const pages = entries.items.map((item) => ({
+    params: { slug: item.fields.slug },
+    props: {
+      title: item.fields.title,
+      content: documentToHtmlString(item.fields.content),
+      date: new Date(item.fields.date).toLocaleDateString(),
+    },
+  }));
+  return pages;
+}
+---
+```
+
+In this example, the `params` property is an object with a `slug` property. The property inside `params` should match the name of the dynamic route (i.e `[slug].astro`).
+
+The `props` property is an object with the properties: title, content and date. We will use the `documentToHtmlString` method to convert the `content` property from a Document to a string of HTML and the Date constructor to format the date.
```

Comment:
Since we already did this, it doesn't make sense to use future tense. Also, the power of a list for readability!
```suggestion
The `props` object passes three properties to the page:
- title (a string)
- content (a rich text Document converted to HTML using `documentToHtmlString`)
- date (formatted using the `Date` constructor)
```